setup.py

```diff
@@ -350,14 +362,20 @@ def awscrt_ext():
             if not is_macos_universal2():
                 extra_link_args += ['-Wl,-fatal_warnings']
 
+    if sys.version_info >= (3, 11):
+        macros.append(('Py_LIMITED_API', '0x030B0000'))
+        py_limited_api = True
+
     return setuptools.Extension(
         '_awscrt',
         language='c',
         libraries=libraries,
         sources=glob.glob('source/*.c'),
         extra_compile_args=extra_compile_args,
         extra_link_args=extra_link_args,
-        extra_objects=extra_objects
+        extra_objects=extra_objects,
+        define_macros=macros,
+        py_limited_api=py_limited_api,
```

Comment:
utterly trivial: rename ~macros~ -> `define_macros`
so it lines up with this function's arg name, just like we're doing with all the other args